tests/framework/results/test_context.py

```diff
@@ -326,15 +345,15 @@ def test_adding_observation_gather_results(
     for result, _measure, _updater in ctx.gather_results(
         population, lifecycle_phase, mocked_event
     ):
-        assert all(
+        assert result is not None and all(
```

Comment:
I think it would be better to have an assert for result is not None and a separate one that does the fuzzy checker instead of the math isclose but I guess that is not really the point of the pr

Reply:
I agree - will update